data/lua/wml-tags.lua

```diff
@@ -911,11 +911,10 @@ function wml_actions.role(cfg)
 
 	filter.role, filter.type = nil, nil
 	local search_map, search_recall, reassign = true, true, true
-	if cfg.search_recall_list ~= nil then
```

Comment:
Did you test that this works? It's possible that `== "no"` will always be false no matter what. I'd recommend testing for "only" and otherwise using the double-not trick.